### PR TITLE
feat(TCOMP-2407) deploy dev branch with qualifier from CI

### PR DESCRIPTION
**Pull request by acatoire**
### Why this PR is needed?
Be able to have a jars and component server image published before merging into master

### What does this PR adds (design/code thoughts)?
In addition, the CI have also been improved:
- split stage to clarify issues source when happening
- improve the Jenkins debug option (aligned with connectors)
- add method to write job description

### CODE REVIEW

File: .jenkins/scripts/docker_push.sh
Comment (undx) [inline .jenkins/scripts/docker_push.sh:1]: Docker images are pushed to the same registry, why do we have two almost identical scripts?

File: .jenkins/scripts/docker_push.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# $3: docker registry to push on
+
+_IMAGE="${1?Missing image name}"
+_TAG="${2?Missing tag}"
+_REGISTRY="${3?Missing registry}"
+
+
+echo ">> Pushing ${_IMAGE}:${_TAG} to ${_REGISTRY}"
+
+mvn install jib:build@build \
```
Comment (undx) [inline .jenkins/scripts/docker_push.sh:32]: why `install`?

File: .jenkins/scripts/docker_push.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    --file "images/${_IMAGE}-image/pom.xml" \
+    --define image.currentVersion="${_TAG}" \
+    --define talend.server.image.registry="${_REGISTRY}" \
+    --activate-profiles release \
+    -DskipTests -Dinvoker.skip=true -T1C
```
Comment (undx) [inline .jenkins/scripts/docker_push.sh:37]: `release`?

File: Jenkinsfile [hunk 3]
```diff
@@ -166,51 +280,96 @@ pipeline {
...
             steps {
                 script {
                     configFileProvider([configFile(fileId: 'maven-settings-nexus-zl', variable: 'MAVEN_SETTINGS')]) {
-                        sh """\
-                            #!/usr/bin/env bash 
-                            bash .jenkins/scripts/docker_build.sh ${env.PROJECT_VERSION}${buildTimestamp}
+                        sh """
+                              bash .jenkins/scripts/docker_build.sh "component-server" ${qualifiedVersion}${buildTimestamp}
+                              bash .jenkins/scripts/docker_build.sh "component-starter-server" ${qualifiedVersion}${buildTimestamp}
+                              bash .jenkins/scripts/docker_build.sh "remote-engine-customizer" ${qualifiedVersion}${buildTimestamp}
```
Comment (undx) [inline Jenkinsfile:335]: launching 3 times the script? older solution spawned only one exec...

File: Jenkinsfile [hunk 4]
```diff
@@ -319,17 +504,13 @@ pipeline {
                         configFileProvider([configFile(fileId: 'maven-settings-nexus-zl', variable: 'MAVEN_SETTINGS')]) {
                             sh """\
                                #!/usr/bin/env bash
-                               bash .jenkins/scripts/release.sh ${env.BRANCH_NAME} ${env.PROJECT_VERSION} 
+                               bash .jenkins/scripts/release.sh ${env.BRANCH_NAME} ${qualifiedVersion}
```
Comment (undx) [inline Jenkinsfile:507]: release script is not modified and this script calls the docker images creation.

File: pom.xml [hunk 5]
```diff
@@ -880,6 +880,16 @@
...
+    <repository>
+      <releases>
+        <enabled>false</enabled>
+      </releases>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+      <id>talend.snapshots</id>
+      <url>https://artifacts-zl.talend.com/nexus/content/repositories/snapshots/</url>
+    </repository>
```
Comment (undx) [inline pom.xml:892]: is that needed?

File: .jenkins/scripts/docker_push.sh (hunk 1 above)
Comment (acatoire) [reply .jenkins/scripts/docker_push.sh:32]: Taken from the post release script
https://github.com/Talend/component-runtime/blob/6a92dcdd310608c5df9c65f742760abc373cd9a4/.jenkins/scripts/post-release.sh#L52-L54

File: pom.xml (hunk 5 above)
Comment (acatoire) [reply pom.xml]: Apparently not, removed

**Review by undx — APPROVED** [review]

**Pull request merged**